devtools/working-with-dsproxy/working-with-dsproxy.md

```diff
@@ -0,0 +1,488 @@
+        uint mkrFee = wdiv(rmul(wad, rdiv(tub.rap(cup), tub.tab(cup))), uint(val));
+
+        uint ethAmt = mkrEx.getEthToTokenOutputPrice(mkrFee);
+        uint daiAmt = daiEx.getTokenToEthOutputPrice(ethAmt);
+
+        daiAmt = add(wad, daiAmt);
+        require(dai.transferFrom(msg.sender, address(this), daiAmt));
+
+        if(ok && val != 0) {
+           daiEx.tokenToTokenSwapOutput(mkrFee, daiAmt, uint(999000000000000000000), uint(1645118771), address(mkr));
+        }
+
+        tub.wipe(cup, wad);
+    }
+}
+
+```
+
+### Deployment and Execution
+
+Before we begin, ensure you have some Kovan ETH to pay gas for transactions and Kovan Dai on the address by following instructions on this [guide](https://github.com/makerdao/developerguides/blob/master/dai/dai-token/dai-token.md#testnet)
+
+Build the `wipe-proxy` project
+
+```bash
+dapp build
+```
+
+Deploy the WipeProxy contract
+
+```bash
+dapp create WipeProxy
+```
+
+Make a note of the contract address returned after successful execution and store it as a variable
+
+```bash
+export WIPEPROXY=0xfd92bd57d369714f519c3e6095d62d5872114e34
+```
+
+Deploy your own DSProxy contract for your address using the factory contract present on Kovan
+
+```bash
+seth send 0x64a436ae831c1672ae81f674cab8b6775df3475c 'build()'
+```
+
+This transaction might fail if you already have deployed a DSProxy contract before from this address. You can check if you have one now with this command
+
+```bash
+seth call 0x64a436ae831c1672ae81f674cab8b6775df3475c 'proxies(address)(address)' 0xYourAddressHere
+```
+
+Make a note of the returned DSProxy contract address to use in a command later.
+
+We can prepare calldata to wipe 1 DAI in debt from CDP #44 on Kovan using this command with the following inputs,
+
+* Address of the Tub contract on Kovan
+* Address of the Uniswap DAI Exchange
+* Address of the Uniswap MKR Exchange
+* CDP #44 in bytes32 format
+* 1 Dai to wipe
+
+```bash
+seth calldata 'wipeWithDai(address,address,address,uint,uint)' 0xa71937147b55deb8a530c7229c442fd3f31b7db2 0x47D4Af3BBaEC0dE4dba5F44ae8Ed2761977D32d6 0x88f55896d822E2355760648731778f21952693AB $(seth --to-hexdata $(seth --to-uint256 44)) $(seth --to-uint256 $(seth --to-wei 1 eth))
+```
+
+Make a note of the returned hex data to use as input while interacting with the DSProxy contract and store it
+
+```bash
+export CALLDATA=0xf4bd4298000000000000000000000000a71937147b55deb8a530c7229c442fd3f31b7db200000000000000000000000047d4af3bbaec0de4dba5f44ae8ed2761977d32d600000000000000000000000088f55896d822e2355760648731778f21952693ab000000000000000000000000000000000000000000000000000000000000002c0000000000000000000000000000000000000000000000000de0b6b3a7640000
+```
+
+Call execute on the DSProxy contract with these inputs,
+
+* Address of the deployed `WipeProxy` contract
+* Calldata to execute the `wipeWithDai` script
+
+```bash
+seth send 0xd5282809a9d40311e06e0992b773075bbd40dde5 'execute(address,bytes memory)' $WIPEPROXY $CALLDATA
```

Comment:
```suggestion
seth send $MYPROXY 'execute(address,bytes memory)' $WIPEPROXY $CALLDATA
```

Reply:
See above comment with respect to $MYPROXY

Reply:
Unfortunately, I'm failing at this line :(
https://kovan.etherscan.io/tx/0x90f386a0fdb74f4977d3149ce727c9fba612497f22247a9f711a6d50517bf3bf